pkg/collector/corechecks/orchestrator/ecs/ecs.go

```diff
@@ -0,0 +1,140 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build orchestrator
+
+// Package ecs is used for the orchestrator ECS check
+package ecs
+
+import (
+	"math/rand"
+	"time"
+
+	"go.uber.org/atomic"
+
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/cluster/orchestrator/collectors"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/cluster/orchestrator/collectors/ecs"
+	oconfig "github.com/DataDog/datadog-agent/pkg/orchestrator/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+)
+
+// CheckName is the name of the check
+const CheckName = "orchestrator_ecs"
+
+// Check doesn't need additional fields
+type Check struct {
+	core.CheckBase
+	WorkloadmetaStore workloadmeta.Component
+	sender            sender.Sender
+	config            *oconfig.OrchestratorConfig
+	collectors        []collectors.Collector
+	groupID           *atomic.Int32
+
+	// isECSCollectionEnabledFunc is used for testing
+	isECSCollectionEnabledFunc func() bool
```

Comment:
could we instead mutate the check config in the test case?

Reply:
I removed this one